tests/check_framework/test_caches.py

```diff
@@ -89,3 +91,24 @@ def test_cache_path_not_conflict(self):
             settings = self.get_settings(setting, root / 'cache', root / 'other')
             with self.subTest(setting=setting), self.settings(**settings):
                 self.assertEqual(check_cache_location_not_exposed(None), [])
+
+class CheckCacheAbsolutePath(SimpleTestCase):
+    warning_message = (
+        "Your '%s' LocalBasedCache LOCATION specifies a relative directory. "
+        "Use an absolute directory to make it independent of the working directory "
+        "of the Django process."
+    )
+
+    @override_settings()
+    def test_no_caches(self):
+        self.assertEqual(check_file_based_cache_is_absolute(None), [])
+
+    @override_settings(CACHES={'default': {'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache', 'LOCATION': '/tmp/django_cache'}})
+    def test_absolute_directory(self):
+        self.assertEqual(check_file_based_cache_is_absolute(None), [])
+
+    @override_settings(CACHES={'default': {'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache', 'LOCATION': 'django_cache'}})
+    def test_relative_directory(self):
+        msg = self.warning_message % ('default', )
+        self.assertEqual(check_file_based_cache_is_absolute(None), [Warning(msg, id='caches.W003')])
```

Comment:
There is no need to declare `warning_message` or `msg`:
```suggestion
        self.assertEqual(check_file_based_cache_is_absolute(None), [
            Warning(   
                "Your 'default' ...", 
                id='caches.W003',
            ),
        ])
```